src/harness/testcases/WINNF_FT_S_SCS_testcase.py

```diff
@@ -257,117 +256,63 @@ def test_WINNF_FT_S_SCS_15(self,config_filename):
     # Check registration response
     self.assertEqual(response['response']['responseCode'],104)
 
-  def generate_SCS_17_default_config(self, filename):
-    """Generates the WinnForum configuration for SCS_17. """
-    # Create the configuration for short lived invalid client cert/key,wait timer
-
-    # Get current working directory
-    cwd = os.getcwd()
-
-    # testcases directory
-    testcases_dir = os.path.dirname(
-      os.path.abspath(inspect.getfile(inspect.currentframe())))
-
-    # Absolute path to the certs directory
-    path = os.path.join(testcases_dir, '../certs')
-
-    # Change directory to the certs directory
-    os.chdir(path)
-
-    # Build short lived certificate command
-    command = "./generate_short_lived_certs.sh  CBSD  short_lived_client " + " 2 "
-
-    # Create the short lived certificate
-    self.assertEqual(os.system(command),0,"short lived certificate creation failed")
-
-    # Revert back to the previous directory
-    os.chdir(cwd)
-
-    config = {
-        'clientCert': self.getCertFilename("short_lived_client.cert"),
-        'clientKey': self.getCertFilename("short_lived_client.key"),
-        'waitTimer': 130
-    }
-    writeConfig(filename, config)
-
-  @configurable_testcase(generate_SCS_17_default_config)
-  def test_WINNF_FT_S_SCS_17(self,config_filename):
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_17(self):
     """Invalid certificate following an approved registration request
        Checks that SAS UUT response with fatal alert message.
     """
-    # Read the configuration
-    config = loadConfig(config_filename)
+    device_cert_name = "short_lived_client"
+    expiry_timer = 2  # in minutes
+
+    self.create_short_lived_certificate("CBSD", device_cert_name, expiry_timer)
+
+    device_cert = self.getCertFilename(device_cert_name + ".cert")
+    device_key = self.getCertFilename(device_cert_name + ".key")
+    wait_timer = (expiry_timer * 60) + 10  # In seconds
 
     # Successful Tls Handshake
     self.assertTlsHandshakeSucceed(self._sas_admin._base_url, ['AES128-GCM-SHA256'],
-                                   config['clientCert'], config['clientKey'])
+                                   device_cert, device_key)
     logging.info("TLS Handshake Succeeded")
 
     # Load the device_a file
     device_a = json.load(
         open(os.path.join('testcases', 'testdata', 'device_a.json')))
 
     with security_testcase.CiphersOverload(self._sas, ['AES128-GCM-SHA256'],
```

Comment:
This feels like overkill. Can you just use the normal registration function?

Reply:
This   approach is used so that the device_cert and device_key configured for this   test case is used. The same approach is used for SCS_18 test case where   assertRegisteredAndGranted method is used with the test case specific device   certificate and key instead of the default ones. This approach follows the   ones used in test cases SCS1 to SCS5.

Reply:
This approach is used for those (SCS.1-5) test cases because we need to specifically test the ciphersuite properties of the SAS. This is not the case in this test.

But I see your point that this is the easiest way to use the existing helper functions without further modification. Please use the original list of ciphersuites and add a note about why you are doing it this way.

Reply:
We   agree. We have added a comment to clarify. Made changes to use the original   list of ciphersuites instead of hardcoding it to   "['AES128-GCM-SHA256']".